### ISSUE TITLE
SemanticsService.announce can fail to announce the message on iOS

### ISSUE THREAD

**Issue by mattrussell-sonocent** [flutter/flutter#52409]
## Steps to Reproduce

1. Run `flutter create bug`.
2. Update the files as follows: 
```
import 'package:flutter/material.dart';
import 'package:flutter/rendering.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      home: Scaffold(
        body: RaisedButton(
          child: Text("Press me"),
          onPressed: () {
            SemanticsService.announce(
                "I've just been pressed", TextDirection.ltr);
          },
        ),
      ),
    );
  }
}
```
3. 
Run on an iOS device with VoiceOver activated.  Ensure button is focused, and then double tap to activate the button. Then repeat.
**Expected results:** 
The screenreader should announce "I've just been pressed" on every button activation.

**Actual results:**
Most of the time, the announcement is lost, and the screenreader just says the button title, "Press me".

<details>
  <summary>Logs</summary>

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, v1.12.13+hotfix.8, on Mac OS X 10.15.2 19C57, locale en-GB)

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
[✓] Android Studio (version 3.5)
[!] IntelliJ IDEA Ultimate Edition (version 2019.3.1)
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
[!] VS Code (version 1.42.1)
    ✗ Flutter extension not installed; install from
      https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter
[✓] Connected device (1 available)

! Doctor found issues in 2 categories.
```

</details>

**Comment by goderbauer** [comment]
This is working as expected. When a button has been pressed, iOS reads out the label of that button and that may interrupt any ongoing announcement. There is nothing Flutter can do about that. If you really wanted to, you could delay the announcement, but I would recommend that.

**Comment by mattrussell-sonocent** [comment]
@goderbauer Thanks for the update, and I appreciate there are limits to what Flutter can do on a particular platform.

> If you really wanted to, you could delay the announcement, but I would recommend that.

Sorry, did you mean "...and I would recommend that" or "...but I wouldn't recommend that"?

If the latter, is there a recommended pattern for announcing an update to screen-reader users in response to an action? In our actual use case, we show a visual message for a few seconds, and we would like to ensure that screen-reader users also get notified.

**Comment by goderbauer** [comment]
It's suppose to say "would not recommend that".
